server/service/apple_mdm.go

```diff
@@ -912,14 +905,78 @@ func enqueueMDMAppleCommandEndpoint(ctx context.Context, request interface{}, sv
 
 func (svc *Service) EnqueueMDMAppleCommand(
 	ctx context.Context,
-	command *fleet.MDMAppleCommand,
+	rawBase64Cmd string,
 	deviceIDs []string,
 	noPush bool,
 ) (status int, result *fleet.CommandEnqueueResult, err error) {
-	if err := svc.authz.Authorize(ctx, command, fleet.ActionWrite); err != nil {
+	var premiumCommands = map[string]bool{
+		"EraseDevice": true,
+		"DeviceLock":  true,
+	}
+
+	// load hosts (lite) by uuids, check that the user has the rigts to run
+	// commands for every affected team.
+	if err := svc.authz.Authorize(ctx, &fleet.Host{}, fleet.ActionList); err != nil {
 		return 0, nil, ctxerr.Wrap(ctx, err)
 	}
-	return deprecatedRawCommandEnqueue(ctx, svc.mdmStorage, svc.mdmPushService, command.Command, deviceIDs, noPush, svc.logger)
+
+	vc, ok := viewer.FromContext(ctx)
+	if !ok {
+		return 0, nil, fleet.ErrNoContext
+	}
+	// for the team filter, we don't include observers as we require maintainer
+	// and up to run commands.
+	filter := fleet.TeamFilter{User: vc.User, IncludeObserver: false}
+	hosts, err := svc.ds.ListHostsLiteByUUIDs(ctx, filter, deviceIDs)
+	if err != nil {
+		return 0, nil, err
+	}
+
+	// collect the team IDs and verify that the user has access to run commands
+	// on all affected teams.
+	teamIDs := make(map[uint]bool)
+	for _, h := range hosts {
+		var id uint
+		if h.TeamID != nil {
+			id = *h.TeamID
+		}
+		teamIDs[id] = true
+	}
+
+	var command fleet.MDMAppleCommand
+	for tmID := range teamIDs {
+		command.TeamID = &tmID
+		if tmID == 0 {
+			command.TeamID = nil
+		}
+
+		if err := svc.authz.Authorize(ctx, command, fleet.ActionWrite); err != nil {
+			return 0, nil, ctxerr.Wrap(ctx, err)
+		}
+	}
+
+	rawXMLCmd, err := base64.RawStdEncoding.DecodeString(rawBase64Cmd)
+	if err != nil {
+		return 0, nil, ctxerr.Wrap(ctx, err, "decode base64 command")
+	}
+	cmd, err := mdm.DecodeCommand(rawXMLCmd)
+	if err != nil {
+		return 0, nil, ctxerr.Wrap(ctx, err, "decode plist command")
+	}
+
+	if premiumCommands[strings.TrimSpace(cmd.Command.RequestType)] {
+		lic, err := svc.License(ctx)
+		if err != nil {
+			return 0, nil, ctxerr.Wrap(ctx, err, "get license")
+		}
+		if !lic.IsPremium() {
+			return 0, nil, fleet.ErrMissingLicense
+		}
+	}
+
+	// TODO(mna): hmmm is this ok to still call this (deprecated) func? Or is
+	// there a ticket to re-implement or address this?
+	return deprecatedRawCommandEnqueue(ctx, svc.mdmStorage, svc.mdmPushService, cmd, deviceIDs, noPush, svc.logger)
```

Comment:
I think Roberto you were the one to rename it `deprecated` ? Does it still make sense to call it? If not, is it planned to be re-implemented?

Reply:
I did 😅 , main reason was to discourage ourselves from using this method, and use `commander.enqueue` instead, which is much, much simpler and doesn't deal with status codes.

Having that said, maybe we can rename it again and revisit if/when we need? seems to work fine for the CLI flow?

Reply:
To actually answer your question, I don't remember creating an issue, and after a quick search I couldn't find anything.

Reply:
> main reason was to discourage ourselves from using this method, and use commander.enqueue instead, which is much, much simpler and doesn't deal with status codes.

@roperzh Gotcha, sounds like it's worth trying to use `commander.Enqueue` instead of this, then? I'll give it a shot, if it's a near drop-in replacement it should be straightforward.

Reply:
sounds good, but if it turns out to be tricky, I have no problems reusing this! I was mostly concerned about internal operations, like sending install/remove profile commands, etc. The return value of this method is very tricky to parse if you're not using it in a HTTP handler.

Reply:
It now uses the `commander.enqueue` method, and I cleaned up (and documented) the endpoint a bit as far as return values goes, since we don't need all that and the commander does not provide all of it either.